llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -2621,6 +2623,20 @@ static Constant *ConstantFoldLibCall2(StringRef Name, Type *Ty,
   switch (Func) {
   default:
     break;
+  case LibFunc_ldexp:
+  case LibFunc_ldexpf:
+  case LibFunc_ldexpl:
+  case LibFunc_scalbn:
+  case LibFunc_scalbnf:
+  case LibFunc_scalbnl:
+  case LibFunc_scalbln:
+  case LibFunc_scalblnf:
+  case LibFunc_scalblnl:
+    if (TLI->has(Func)) {
+      APFloat ret = llvm::scalbn(Op1V, (int)Op2V.convertToDouble(), RoundingMode::TowardZero);
```

Comment:
You shouldn't need to convert. The second argument is always an in integer (such that this code is unreachable, the APFloat won't match)